app/scripts/session/sessionEvent.js

```diff
@@ -41,6 +41,10 @@ window.vcSession = window.vcSession ? window.vcSession : {};
             var _selectedCases = vcSession.utils.buildCaseListObject(
               _studyMatch.selectedCases, cancerStudyID, sampleID);
             _studyMatch.selectedCases = _selectedCases;
+
+            // TODO: this is the question I have for a while, should we have
+            // individual length property? I understand it's convenient
+            // but also easy to get out of sync with the samples array.
```

Comment:
Good catch. We should get rid of both samplesLength and patientsLength. And them from the array's.
For this we need to change login in both this module and iviz module. Will work on this